### PR TITLE
updated photonlib and robot pose estimator docs

### PR DESCRIPTION
**Pull request by mdurrani808**

### CODE REVIEW

File: source/docs/programming/photonlib/robot-pose-estimator.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Creating an ``AprilTagFieldLayout``
+-----------------------------------
+``AprilTagFieldLayout`` is used to represent a layout of AprilTags within a space (field, shop at home, classroom, etc.). WPILib provides a JSON that describes the layout of AprilTags on the field which you can then use in the AprilTagFieldLayout constructor. You can also specify a custom layout.
+
+The API documentation can be found in here: `Java <https://github.wpilib.org/allwpilib/docs/beta/java/edu/wpi/first/apriltag/AprilTagFieldLayout.html>`_ and `C++ <https://github.wpilib.org/allwpilib/docs/beta/cpp/classfrc_1_1_april_tag_field_layout.html>`_.
+
+.. tab-set-code::
+   .. code-block:: java
+
+      AprilTagFieldLayout aprilTagFieldLayout = new ApriltagFieldLayout("PATH/TO/LAYOUT/JSON.json");
```
Comment (mcm001) [inline source/docs/programming/photonlib/robot-pose-estimator.rst:17]: This should be bundled with wpilib, people shouldn't need to download it.

File: source/docs/programming/photonlib/robot-pose-estimator.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+   .. code-block:: c++
+
```
Comment (mcm001) [inline source/docs/programming/photonlib/robot-pose-estimator.rst:20]: todo

File: source/docs/programming/photonlib/robot-pose-estimator.rst (hunk 1 above)
Comment (mcm001) [reply source/docs/programming/photonlib/robot-pose-estimator.rst:17]: `AprilTagFieldLayout.loadFromResource(AprilTagFields.k2022RapidReact.m_resourceFile);`

File: source/docs/programming/photonlib/using-target-data.rst [hunk 3]
```diff
@@ -55,3 +69,29 @@ You can get your robot's ``Pose2D`` on the field using various camera data, targ
...
+``estimateFieldToRobotAprilTag(Transform3d cameraToTarget, Pose3d fieldRelativeTagPose, Transform3d cameraToRobot)`` returns your robot's ``Pose3d`` on the field using the pose of the AprilTag relative to the camera, pose of the AprilTag relative to the field, and the transform from the camera to the origin of the robot.
+
+.. tab-set-code::
+   .. code-block:: java
+
+      // Calculate robot's field relative pose
+      Pose3D robotPose = PhotonUtils.estimateFieldToRobotAprilTag(target.getBestCameraToTarget(), aprilTagFieldLayout.getTagPose(target.getFiducialId()), cameraToRobot);
+   .. code-block:: c++
+
+     //TODO
```
Comment (mcm001) [inline source/docs/programming/photonlib/using-target-data.rst:84]: todo

File: source/docs/programming/photonlib/using-target-data.rst [hunk 4]
```diff
@@ -55,3 +69,29 @@ You can get your robot's ``Pose2D`` on the field using various camera data, targ
...
+
+Estimating Field Relative Pose with AprilTags
+---------------------------------------------
+``estimateFieldToRobotAprilTag(Transform3d cameraToTarget, Pose3d fieldRelativeTagPose, Transform3d cameraToRobot)`` returns your robot's ``Pose3d`` on the field using the pose of the AprilTag relative to the camera, pose of the AprilTag relative to the field, and the transform from the camera to the origin of the robot.
+
+.. tab-set-code::
+   .. code-block:: java
+
+      // Calculate robot's field relative pose
+      Pose3D robotPose = PhotonUtils.estimateFieldToRobotAprilTag(target.getBestCameraToTarget(), aprilTagFieldLayout.getTagPose(target.getFiducialId()), cameraToRobot);
```
Comment (mcm001) [inline source/docs/programming/photonlib/using-target-data.rst:81]: Do we need to check that the target ID is contained in the field layout before doing this?

Comment (mdurrani808) [reply source/docs/programming/photonlib/using-target-data.rst:81]: There is no way to easily do this with AprilTag Field layout so not really sure.

Comment (mcm001) [reply source/docs/programming/photonlib/using-target-data.rst:81]: It should return Optional?

**Review by mcm001 — APPROVED** [review]

**Pull request merged**